includes/wc-update-functions.php

```diff
@@ -2242,3 +2218,53 @@ function wc_update_450_sanitize_coupons_code() {
 	delete_option( 'woocommerce_update_450_last_coupon_id' );
 	return false;
 }
+
+/**
+ * Remove the product attribute related terms for variation products.
+ * Those are no longer created, and no longer needed after the new mechanism to improve the filtering of
+ * variable products by attribute was reverted (https://github.com/woocommerce/woocommerce/pull/27625).
+ *
+ * @return bool True if the script needs to be run again.
+ */
+function wc_update_470_remove_term_relationships_for_variations() {
+	global $wpdb;
```

Comment:
let's not remove the terms, in case an extension is using a similar setup for their functionality. although deleting them when the term is unset is likely to be safe. 
tbh, we can keep them as it is (even if stale) unless there are reasons to remove them.

Reply:
Good point. Migration removed.